### DISCUSSION TITLE
[Refactoring] Add ability (in context menu) to automatically define a function from line which calls undefined function

### DISCUSSION THREAD

**Discussion by mystiquewolf** [microsoft/pylance-release#2598] [Enhancement]
When Pylance says that a function is undefined (when having written a line that calls that undefined function), add ability to select a refactoring method which will automatically create a function with signature/parameters based on the calling line.
So that it doesn't say "No quick fixes available"

![Screenshot_20210811_172015](https://user-images.githubusercontent.com/16529989/129046263-43b6c59c-a250-4835-95de-ad8c6d648329.png)

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by heejaechang** [comment]
see https://devblogs.microsoft.com/python/python-in-visual-studio-code-february-2025-release/#ai-code-action:-generate-symbol-(experimental)
